starters/storybook/app/components/primitives/Button/Button.tsx

```diff
@@ -0,0 +1,41 @@
+import { LucideIcon } from 'lucide-react'
+import { Link } from '~/components/primitives'
+import {
+  Button as ShadcnButton,
+  ButtonProps as ShadcnButtonProps,
+} from '~/components/ui/button'
+
+export interface ButtonProps extends Omit<ShadcnButtonProps, 'asChild'> {
+  href: string
+  internal?: boolean
+  text: string
+  icon?: LucideIcon
+}
+
+export const Button = ({ href, text, internal, ...props }: ButtonProps) => {
+  const externalProps = internal
```

Comment:
I saw this logic repeated in the link component, this doesn't belong in that component?